addon/-private/system/model/model.js

```diff
@@ -1631,11 +1498,13 @@ Model.reopenClass({
    @param {any} binding the value to which the callback's `this` should be bound
    */
   eachRelationship(callback, binding) {
-    get(this, 'relationshipsByName').forEach((relationship, name) => {
-      callback.call(binding, name, relationship);
-    });
+    get(this, '__internalModelClass').eachRelationship(callback, binding);
   },
 
+  __internalModelClass: computed(function() {
```

Comment:
I don't think we should introduce another property on DS.Model. This is accessible via `this._internalModel.modelClass`, I think we should use that, rather then the CP indirection.

* it will be quicker then a CP
* less things on the DS.Model the better